### PR TITLE
Implementing CODEOWNERS

### PR DESCRIPTION
**Pull request by eiso**
Suggested by @dpordomingo here https://github.com/src-d/guide/issues/99 and supported by @campoy in https://github.com/src-d/guide/issues/105

This PR implements the CODEOWNERS suggestion from @dpordomingo and reorganizes the repository and its README.md accordingly. 

The rationale is:

* A maintainer per subsection gives more ownership 
* Easier to maintain

Before merging this PR, I'd like the approval @jorgeschnura since you've been the global maintainer so far. You remain the global maintainer but subsections will have owners. Feel free to make any obvious changes directly on this branch.

### CODE REVIEW

**Comment by bzz** [comment]
Looks really nice! 

Seems like a rebase is in order.

File: engineering/methodology.md [hunk 1]
```diff
@@ -108,16 +108,14 @@ Demo meetings
...
   + Starting at the end of the second sprint, one every other sprint.
 
-  + Every 2nd week of sprint, on Fridays after the [sprint demo](https://github.com/src-d/guide/blob/master/engineering/methodology.md#Demo).
+  + Every 2nd week of sprint, on Fridays after the [sprint demo](#Demo).
 
 ### Keeping track of things:
 
   + Publicly: a Kanban board on-line.
 
   + Per team: any way team sees fit (github issues, github projects, waffle.io...).
```
Comment (bzz) [inline engineering/methodology.md]: while we are still here, may be "team sees fit (most teams use github projects)"

Comment (eiso) [reply engineering/methodology.md]: I won't change this since it should be reflected in the PR on methodology

**Comment by eiso** [comment]
LGTM, after in-person discussion with @jorgeschnura

**Pull request merged**